desktop_version/src/Game.cpp

```diff
@@ -4837,6 +4842,7 @@ void Game::serializesettings(tinyxml2::XMLElement* dataNode, const struct Screen
 
     xml::update_tag(dataNode, "lang", loc::lang.c_str());
     xml::update_tag(dataNode, "lang_set", (int) loc::lang_set);
+    xml::update_tag(dataNode, "new_level_font", loc::new_level_font.c_str());
```

Comment:
I'm not sure if this should be saved to game settings as the new default level font. We don't do things like, I don't know, save the default author name to game settings for convenience. I don't see the problem with just letting people set the font to the font they want manually whenever they create new levels (which I can't imagine would be too often, and if you're doing something like creating many test levels, there's still a bunch of other stuff you have to set up each time anyways so I don't see the point). And in fact, this might not make a difference anyways if people are creating levels with custom fonts or a bunch of different fonts (they'd still have to go through the menu and change the font).

Reply:
If someone is for example Japanese, they expect to be able to make a Japanese level. So in a Japanese VVVVVV, the font _has_ to be the Japanese font by default, because they won't immediately know there is an option to change it, or understand why they can't type properly in the editor.

But if you want to always make English levels (which is understandable - I'm Dutch myself, we have people from France, Germany, etc, all not restricting themselves to just communities in their own language, and making English levels instead), having to manually switch to the 8x8 font every time would get annoying, and I do not want to annoy people into setting their entire VVVVVV to English just because they make levels for an international audience.

So, I'd normally be saying the same thing - "just change the font manually whenever you create a new level" - but I really want to avoid making using your preferred language have any annoying downsides that make people switch away from using their preferred language. Same principle will apply to localized LIES/YES/etc enemies: casual gamers might prefer them, but there _has_ to be an option to use the English graphics, otherwise all speedrunners will feel forced to set their _entire game_ to English just so they're not at a disadvantage with the pixel-perfect enemy collision.

Reply:
> So, I'd normally be saying the same thing - "just change the font manually whenever you create a new level" - but I really want to avoid making using your preferred language have any annoying downsides that make people switch away from using their preferred language.

I guess I'm not fully on board with this option being implicitly made the new default, when there's nowhere that explicitly indicates that, and the user might not expect it to be the case. Maybe add some text stating that the selected font will be the new default?

There's also the issue that I'm not sure how to clear the default option and reset it back to the _default_ default (automatically using the font of the currently selected language) without having to manually edit `settings.vvv`. There doesn't appear to be an "automatic" option in the font selection menu.

> Same principle will apply to localized LIES/YES/etc enemies: casual gamers might prefer them, but there _has_ to be an option to use the English graphics, otherwise all speedrunners will feel forced to set their _entire game_ to English just so they're not at a disadvantage with the pixel-perfect enemy collision.

I thought that this wouldn't be an issue in the first place, because no matter what graphics are drawn, the collision would always use the English graphics. Unless the issue is with the mismatched visuals and collision?

Reply:
> I guess I'm not fully on board with this option being implicitly made the new default, when there's nowhere that explicitly indicates that, and the user might not expect it to be the case. Maybe add some text stating that the selected font will be the new default?

I personally don't think it's a huge issue if people make a new level and didn't expect the font to be carried over from the last time they changed it - if anything, they'll go "ah, so I just have to change it back now and _that_ will be remembered". In any case, they already know the menu exists because they used it before, and by seeing text being in the wrong font, it's obvious the game apparently remembers what font they set for their next level.

> There's also the issue that I'm not sure how to clear the default option and reset it back to the _default_ default (automatically using the font of the currently selected language) without having to manually edit `settings.vvv`. There doesn't appear to be an "automatic" option in the font selection menu.

I don't think people need to be aware of when the font is attached to the language and when it isn't (how often do people change a level's font and then change the game's language and then look confused at the font in new levels they make), but if you switch the level font to the current language's font (in other words, you're switching it _back_ to what it was), it reattaches the default font to the current language.

I agree it's not perfect mind-reading, but I think it's the only system that's user-friendly in all cases and does the right thing most of the time (for both inexperienced and impatient users who can't be bothered to read UI text, as well as experienced and attentive users), and it's better than adding specific menu clutter to control what the default should be and being verbose about it, or expecting users to understand text that could easily be misunderstood as "changing the font for this level will change it for all existing levels too, and they will break" or be a hard to understand crammed line that raises about as many questions as "the old level's new font will instead of the old default font now be the new _default_ for any _more_ recent level!". I think it's not too important for people to have to think too hard about it when they need to change the font.

The other option is to always ask what the font should be when creating a new level, but I don't really like that idea, and even then I feel like making a choice should also initialize the selection to that choice the next time you make a new level.

> I thought that this wouldn't be an issue in the first place, because no matter what graphics are drawn, the collision would always use the English graphics. Unless the issue is with the mismatched visuals and collision?

Yeah, I'm talking about the visuals - if you can't "see exactly what you're doing" in your preferred language and you can in English, that'd be an unreasonable downside of using your preferred language for speedrunners.

Reply:
I think the text can just be as simple as "This will also be the default font for new levels you create." It's a whole different menu, there's enough space for it.

Reply:
But then again: why do people need to know upfront? Can't they see it when they make a new level, and put any text in it? If anyone is going to be that confused because they don't remember changing the font for another level, then they won't remember that text either. I don't think it's something people will need to think about that much; if the font doesn't look right, now you know it's being remembered, and you can just change it.

Reply:
I've dealt with way too many users to know just how much they can miss what should be obvious. I'd rather put the text in just to be safe rather than sorry, and I don't think there's any downsides to putting the text in.

Reply:
That is actually exactly my concern: many users will not read and understand the text anyway, or will just miss it, and otherwise they're made to pause and think about the implications of being warned that they're making a change that will affect them in the future. It may read like a warning, something you need to pay attention to, when in fact, I don't think anyone will find it important enough? And it has a kind of "now's the moment to decide, you cannot change it later" air around it, making it feel important.

I don't see what the issue is if someone changes the font, and then later makes a new level, and the editor is still on that font _without_ the user knowing. (Even if the user knows because they read the text, it might not be on their mind.) As soon as they're surprised by the font in their level, they can change it (they've shown they at least know you can), and then it doesn't bother them again for newer levels. Unless they really expect the game to be able to read their mind.

Reply:
Okay. You can keep it this way then.

I do think it can be improved by having text that says which font is used on the previous screen (and in the list too, but especially the previous screen) without needing the user to go in to the list to check. Because some may not realize you can press Esc to go back, and there's no button in the list to go back to the previous menu without selecting a font.

Reply:
Hm, I think that makes sense - it'd be like the slowdown menu, which shows the current setting as well as letting you change it.

I guess on the previous screen it would be something on the bottom left, just like the editor ghosts hint? (Except maybe permanently visible instead of just when you highlight the option, because it _is_ a kind of metadata...)